brainstorming.md

```diff
@@ -0,0 +1,372 @@
+| obs, var                | Field value type is `SOMADataFrame`                                                                                                                                                      |
+| obsm, varm              | Field value type is a `SOMACollection`, where each element in the collection has a value of type `SOMADataFrame`                                                                         |
+| obsp, varp , X          | Field value type is a `SOMACollection`, where each element in the collection has a value of type `SOMAMatrix`                                                                            |
+| obs, obsm               | Index column name is `'obs_id'`. All `obs_id` values exist in the `obs_id` column of the `obs` field.                                                                                    |
+| var, varm               | Index column name is `'var_id'`. All `var_id` values exist in the `var_id` column of the `var` field.                                                                                    |
+| obsp                    | Index column names are `['obs_1_id', 'obs_2_id']`. All `obs_1_id` and `obs_2_id` values exist in the `obs_id` column of the `obs` field.                                                 |
+| varp                    | Index column names are `['var_1_id', 'var_2_id']`. All `var_1_id` and `var_2_id` values exist in the `var_id` column of the `var` field.                                                 |
+| X                       | Index column names are `['obs_id', 'var_id']`. All `obs_id` values exist in the `obs_id` column of the `obs` field. All `var_id` values exist in the `var_id` column of the `var` field. |
+
+> ⚠️ **Issue** `raw` -- do we need `raw.X` and `raw.var` at all, or can we model them with existing structures? **Proposal**: the current `X` and `varm` can represent typical `raw` semantics (filtered subset, or transformed values). Assume that higher level usage conventions will utilize these to represent raw, eg, `X['raw']` and `varm['raw']`, or alternatively by adding additional columns to `var` that indicate filtering status.
+
+> ⚠️ **Issue** `obsm` and `varm` schema -- if these are RecordBatches indexed by obs_id / var_id, then they are essentially identical to obs/var slots. Are they necessary, or can we represent obsm/varm with conventions layered on top of obs/var? if they are different, how? The only current difference is that `obs` and `var` are required to have an entry for each value on that dimension, whereas `obsm`/`varm` do not require all dimension IDs to exist (ie, they are sparse like X). **Proposal**: retain, but differentiate by considering obsm/varm as secondary annotations on the axis, which do **not** require a value for each obs_id/var_id (ie are **sparse**).
```

Comment:
i prefere them separately stored (to ease mental framing by all parties involved) but could be talked out of that with a compelling reason to overlap storage